packages/components/managed-data-table/src/data-table-settings/column-manager/column-manager.js

```diff
@@ -0,0 +1,246 @@
+              <AccessibleButton
+                onClick={props.onClose}
+                label={formatMessage(messages.closeButtonLabel)}
+              >
+                <CloseIcon size="medium" />
+              </AccessibleButton>
+            </HeaderContainer>
+            <Spacings.Stack scale="s">
+              <Spacings.Inset scale="xs">
+                <div style={containerStyles}>
+                  <div ref={registerContentNode}>
+                    <DragDropContext
+                      onDragEnd={(dragResult) =>
+                        handleDragEnd(
+                          dragResult,
+                          props.onUpdateColumns,
+                          props.selectedColumns,
+                          props.availableColumns,
+                          setIsDragging
+                        )
+                      }
+                      onDragStart={handleDragStart}
+                    >
+                      <Spacings.Inline scale="m">
+                        <DroppableContainer isDragging={isDragging}>
+                          <Spacings.Stack>
+                            <Spacings.Inline scale="xs" alignItems="center">
+                              <EyeCrossedIcon size="medium" />
+                              <FieldLabel
+                                title={formatMessage(messages.hiddenColumns)}
+                              />
+                            </Spacings.Inline>
+                            {props.areHiddenColumnsSearchable && (
+                              <AsyncSelectInput
+                                id="hidden-columns-panel-search-input"
+                                {...(props.searchHiddenColumnsPlaceholder
+                                  ? {
+                                      placeholder:
+                                        props.searchHiddenColumnsPlaceholder,
+                                    }
+                                  : undefined)}
+                                cacheOptions={false}
+                                onChange={() => {
+                                  // to avoid prop-types error
+                                }}
+                                // loadOptions is used instead of onInputChange
+                                // because the loading indicator is displayed
+                                // only when loadOptions is invoked
+                                loadOptions={debounce(
+                                  props.searchHiddenColumns,
+                                  300
+                                )}
+                                onInputChange={(input) =>
+                                  // loadOptions is not invoked when input is empty
+                                  !input.length &&
+                                  props.searchHiddenColumns(input)
+                                }
+                                components={{
```

Comment:
Nit: Feels to me we can hoist them up as `const selectInputComponents` to not pass new references.